src/block.js

```diff
@@ -191,7 +191,18 @@ Block.calculateTarget = function (bits) {
   var exponent = ((bits & 0xff000000) >> 24) - 3
   var mantissa = bits & 0x007fffff
   var target = Buffer.alloc(32, 0)
-  target.writeUInt32BE(mantissa, 28 - exponent)
+  if (exponent < 0) {
```

Comment:
Is this the bug that you mentioned that you fixed upstream?

Reply:
Yup